app/src/schemas/messageSchemas.ts

```diff
@@ -146,77 +152,38 @@ export const promptMessagesJSONSchema = zodToJsonSchema(promptMessagesSchema, {
   removeAdditionalStrategy: "passthrough",
 });
 
-/**
- * Conversion Functions
- *
- * These follow a hub-and-spoke model where OpenAI is the hub format.
- * All conversions between different formats go through OpenAI as an intermediate step.
- */
-
-/**
- * Spoke → Hub: Convert an Anthropic message to OpenAI format
- */
-export const anthropicMessageToOpenAI = anthropicMessageSchema.transform(
-  (anthropic): OpenAIMessage => {
-    const base: OpenAIMessage = {
-      role: anthropic.role as OpenAIMessageRole,
-      content: Array.isArray(anthropic.content)
-        ? anthropic.content
-            .filter((block) => block.type === "text" && block.text)
-            .map((block) => block.text!)
-            .join("\n")
-        : anthropic.content,
-    };
-
-    if (anthropic.tool_calls) {
-      return {
-        ...base,
-        tool_calls: anthropic.tool_calls.map((tc) =>
-          anthropicToolCallToOpenAI.parse(tc)
-        ),
-      };
-    }
-
-    if (anthropic.tool_call_id) {
-      return {
-        ...base,
-        tool_call_id: anthropic.tool_call_id,
-      };
-    }
-
-    return base;
-  }
-);
-
 /**
  * Hub → Spoke: Convert an OpenAI message to Anthropic format
  */
 export const openAIMessageToAnthropic = openAIMessageSchema.transform(
   (openai): AnthropicMessage => {
-    const base: AnthropicMessage = {
-      role:
-        openai.role === "system"
-          ? "user"
-          : (openai.role as AnthropicMessageRole),
+    const base = {
+      role: openai.role === "assistant" ? "assistant" : "user",
```

Comment:
does this fallback make sense?

Reply:
My justification is that anthropic only supports assistant and user messages.